tools/build_defs/docker/build.bzl

```diff
@@ -219,12 +219,15 @@ def _compute_layer_name(ctx, layer):
 
 def _repository_name(ctx):
   """Compute the repository name for the current rule."""
+  if ctx.attr.image_name:
+    return os.path.join(ctx.attr.repository, ctx.label.image_name)
+
   if ctx.attr.legacy_repository_naming:
     # Legacy behavior, off by default.
-    return "%s/%s" % (ctx.attr.repository, ctx.label.package.replace("/", "_"))
+    return os.path.join(ctx.attr.repository, ctx.label.package.replace("/", "_"))
```

Comment:
s/os.path.join/_join_path/

I meant the equivalent of this std Python library, which for Bazel is defined in `path.bzl`

The same below.